### PR TITLE
add gcan layers

### PR DESCRIPTION
**Pull request by DavidRuhe**

### CODE REVIEW

File: cliffordlayers/cliffordalgebra.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def geometric_product(self, a, b):
+        return torch.einsum(
+            "bfi,ijk,bfk->bfj",
+            a.view(len(a), -1, len(self.cayley)),
+            self.cayley,
+            b.view(len(b), -1, len(self.cayley)),
+        ).view(*a.size())
+
+    def reverse(self, mv, blades=None):
+        grades = self.bbo.grades.to(mv.device)
```
Comment (rejuvyesh) [inline cliffordlayers/cliffordalgebra.py:43]: Would be super useful to add documentation to these functions of this class! `reverse`, `embed` are fairly generic terms and readers/users may not know what they mean in this context.

File: cliffordlayers/models/gcan_layers.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+        torch.nn.init.kaiming_uniform_(self._weight, a=math.sqrt(5))
+
+    @property
+    def action(self):
+        return self.algebra.embed(self._action, self.action_blades)
+
+    @property
+    def weight(self):
+        return self._weight
```
Comment (rejuvyesh) [inline cliffordlayers/models/gcan_layers.py]: Is it necessary to hide it this way? Why not just do `self.weight` above?

File: cliffordlayers/models/gcan_layers.py
Comment (rejuvyesh) [inline cliffordlayers/models/gcan_layers.py:1]: Should move these layers to `nn/modules`.

File: cliffordlayers/nn/modules/cliffordg3conv.py
Comment (rejuvyesh) [inline cliffordlayers/nn/modules/cliffordg3conv.py:1]: Should move these to `nn/functional`

File: tests/test_gcan_layers.py
Comment (rejuvyesh) [inline tests/test_gcan_layers.py:1]: Can we assert something like expected shapes etc in the tests?

**Review by rejuvyesh — CHANGES_REQUESTED** [review]
It's a great start! Let's refactor and document this a bit!

File: cliffordlayers/cliffordalgebra.py (hunk 1 above)
Comment (DavidRuhe) [reply cliffordlayers/cliffordalgebra.py:43]: Added these!

File: cliffordlayers/models/gcan_layers.py (hunk 2 above)
Comment (DavidRuhe) [reply cliffordlayers/models/gcan_layers.py]: Updated as you suggested.

File: cliffordlayers/models/gcan_layers.py
Comment (DavidRuhe) [reply cliffordlayers/models/gcan_layers.py:1]: Done.

File: cliffordlayers/nn/modules/cliffordg3conv.py
Comment (DavidRuhe) [reply cliffordlayers/nn/modules/cliffordg3conv.py:1]: Done.

File: tests/test_gcan_layers.py
Comment (DavidRuhe) [reply tests/test_gcan_layers.py:1]: Done.

**Comment by DavidRuhe** [comment]
@microsoft-github-policy-service agree

**Pull request merged**